src/librustdoc/html/render/print_item.rs

```diff
@@ -1430,22 +1431,86 @@ fn item_enum(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, e: &clean::
     write!(w, "{}", document(cx, it, None, HeadingOffset::H2));
 
     if count_variants != 0 {
-        item_variants(w, cx, it, e.variants());
+        item_variants(w, cx, it, &e.variants);
     }
     let def_id = it.item_id.expect_def_id();
     write!(w, "{}", render_assoc_items(cx, it, def_id, AssocItemRender::All));
     write!(w, "{}", document_type_layout(cx, def_id));
 }
 
-fn render_enum_fields<'a>(
+fn get_parent_enum_def_id(
+    cx: &mut Context<'_>,
+    variant_def_id: rustc_hir::def_id::LocalDefId,
+) -> DefId {
+    use rustc_hir::{ItemKind, Node};
+    let variant_hir_id = cx.tcx().hir().local_def_id_to_hir_id(variant_def_id);
+
+    for (_, node) in cx.tcx().hir().parent_iter(variant_hir_id) {
+        if let Node::Item(item) = node && matches!(item.kind, ItemKind::Enum(..)) {
+            return item.owner_id.to_def_id();
+        }
+    }
+    panic!("No parent enum found for variant {variant_def_id:?}");
+}
+
+/// It'll return true if all variants are C-like variants and if at least one of them has a value
+/// set.
+fn should_show_c_like_variants_value(
+    variants: &rustc_index::IndexVec<rustc_target::abi::VariantIdx, clean::Item>,
+) -> bool {
+    let mut nb_c_like_variants_with_value = 0;
+    for variant in variants {
+        if let clean::VariantItem(ref var) = *variant.kind &&
+            matches!(var.kind, clean::VariantKind::CLike)
+        {
+            if var.discriminant.is_some() {
+                nb_c_like_variants_with_value += 1;
+            }
+        } else {
+            return false;
+        }
+    }
+    nb_c_like_variants_with_value > 0
+}
+
+fn display_c_like_variant(
+    w: &mut Buffer,
+    cx: &mut Context<'_>,
+    item: &clean::Item,
+    variant: &clean::Variant,
+    index: rustc_target::abi::VariantIdx,
+    should_show_c_like_variants_value: bool,
+) {
+    let name = item.name.unwrap();
+    if let Some(ref value) = variant.discriminant {
+        write!(w, "{} = {}", name.as_str(), value.value(cx.tcx(), true));
+    } else if should_show_c_like_variants_value &&
+        let Some(variant_def_id) = item.item_id.as_def_id() &&
+        let Some(variant_def_id) = variant_def_id.as_local()
+    {
+        let enum_def_id = get_parent_enum_def_id(cx, variant_def_id);
```

Comment:
Due to `get_parent_enum_def_id()` requiring a `LocalDefId` and access to the HIR, we don't render the discriminant of cross-crate re-exported enum variants. I haven't checked if it works but can't we obtain the enum `DefId` via `tcx.parent()` instead?

```suggestion
        let Some(variant_def_id) = item.item_id.as_def_id()
    {
        let enum_def_id = cx.tcx.parent(variant_def_id);
```

And in case that doesn't pan out, could you somehow pass the enum `DefId` as parameter to `display_c_like_variant`?

Reply:
I think it's possible (and it would also allow to not need this function anymore). Let me double-check.

Reply:
Confirmed it can be done, much better this way!